extensions/amp-consent/0.1/consent-info.js

```diff
@@ -191,36 +212,55 @@ export function isConsentInfoStoredValueSame(infoA, infoB, opt_isDirty) {
     } else {
       isDirtyEqual = !!infoA['isDirty'] === !!infoB['isDirty'];
     }
-    return stateEqual && stringEqual && isDirtyEqual;
+    const metadataEqual = isMetadataEqual(
+      infoA['consentMetadata'],
+      infoB['consentMetadata']
+    );
+    return stateEqual && stringEqual && metadataEqual && isDirtyEqual;
   }
   return false;
 }
 
+/**
+ * Compares metadata values from consentInfo
+ *
+ * @param {Object=} unusedMetadataA
+ * @param {Object=} unusedMetadataB
+ * @return {boolean}
+ */
+function isMetadataEqual(unusedMetadataA, unusedMetadataB) {
+  return true;
+}
+
 /**
  * Convert the legacy boolean stored value to consentInfo object
  * @param {boolean} value
  * @return {!ConsentInfoDef}
  */
 function getLegacyStoredConsentInfo(value) {
   const state = convertValueToState(value);
-  return constructConsentInfo(state, undefined, undefined);
+  return constructConsentInfo(state);
 }
 
 /**
  * Construct the consentInfo object from values
+ *
  * @param {CONSENT_ITEM_STATE} consentState
  * @param {string=} opt_consentString
+ * @param {Object=} opt_consentMetadata
  * @param {boolean=} opt_isDirty
  * @return {!ConsentInfoDef}
  */
 export function constructConsentInfo(
   consentState,
   opt_consentString,
+  opt_consentMetadata,
```

Comment:
I understand why you want to pass the consent metadata before the `isDirty`. Please double check that all calling to the `#constructConsentInfo()` have been updated.

Reply:
Yup. Done